extensions/panache/hibernate-orm-panache-rest/deployment/src/main/java/io/quarkus/hibernate/orm/panache/rest/deployment/methods/UpdateMethodImplementor.java

```diff
@@ -0,0 +1,108 @@
+        ControllerAnnotator.addLinks(methodCreator, getEntityType(), "update");
+
+        BranchResult entityDoesNotExist = methodCreator
+                .ifNull(findEntity(methodCreator, methodCreator.getMethodParam(0), getEntityType()));
+
+        createAndReturn(entityDoesNotExist.trueBranch());
+        updateAndReturn(entityDoesNotExist.falseBranch());
+
+        methodCreator.close();
+    }
+
+    @Override
+    protected void implementHal(ClassCreator classCreator) {
+        MethodCreator methodCreator = classCreator.getMethodCreator("updateHal", Response.class, getIdType(), getEntityType());
+        ControllerAnnotator.addTransactional(methodCreator);
+        ControllerAnnotator.addPut(methodCreator);
+        ControllerAnnotator.addPath(methodCreator, EntityUrlProvider.getCollectionUrl(getEntityType()) + "/{id}");
+        ControllerAnnotator.addPathParam(methodCreator.getParameterAnnotations(0), "id");
+        ControllerAnnotator.addConsumes(methodCreator, ControllerAnnotator.APPLICATION_JSON);
+        ControllerAnnotator.addProduces(methodCreator, ControllerAnnotator.APPLICATION_HAL_JSON);
+
+        BranchResult entityDoesNotExist = methodCreator
+                .ifNull(findEntity(methodCreator, methodCreator.getMethodParam(0), getEntityType()));
+
+        createAndReturnHal(entityDoesNotExist.trueBranch());
+        updateAndReturn(entityDoesNotExist.falseBranch());
+
+        methodCreator.close();
+    }
+
+    private void createAndReturn(BytecodeCreator creator) {
+        ResultHandle entity = merge(creator, creator.getMethodParam(0), creator.getMethodParam(1));
+        ResultHandle url = urlProvider.getEntityUrl(creator, entity, getEntityType());
+        ResultHandle response = ControllerResponseFactory.created(creator, entity, url);
+        creator.returnValue(response);
+    }
+
+    private void createAndReturnHal(BytecodeCreator creator) {
+        ResultHandle entity = merge(creator, creator.getMethodParam(0), creator.getMethodParam(1));
+        ResultHandle url = urlProvider.getEntityUrl(creator, entity, getEntityType());
+        ResultHandle response = ControllerResponseFactory.created(creator, wrapHalEntity(creator, entity), url);
+        creator.returnValue(response);
+    }
+
+    private void updateAndReturn(BytecodeCreator creator) {
+        merge(creator, creator.getMethodParam(0), creator.getMethodParam(1));
+        creator.returnValue(ControllerResponseFactory.noContent(creator));
+    }
+
+    private ResultHandle merge(BytecodeCreator creator, ResultHandle id, ResultHandle entity) {
+        fieldAccessor.setId(creator, getEntityType(), entity, id);
+        ResultHandle entityManager = creator.invokeStaticMethod(
+                MethodDescriptor.ofMethod(JpaOperations.class, "getEntityManager", EntityManager.class));
+        return creator.invokeInterfaceMethod(
+                MethodDescriptor.ofMethod(EntityManager.class, "merge", Object.class, Object.class), entityManager, entity);
```

Comment:
@FroMage I cannot resist but to ask you the question. Here `merge` is used and I know you are against its usage, so should this PR provides an other implementation ?

Reply:
Pfff… Well yeah, this feature uses the entities as DTOs which we're advising against, but OTOH we don't have a better solution yet, so probably this has to be allowed here until we have time to come up with our DTO story.